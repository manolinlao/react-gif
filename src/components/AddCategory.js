import React, { useState } from 'react'

export const AddCategory = ( { setCategories } ) => {

  const [inputValue,setInputValue] = useState('');

  const handleAdd = () => {
    if(inputValue.trim().length>2){
      setCategories( cats => [...cats,inputValue] );
      setInputValue('');
    }   
  }

  const handleInputChange = ( e ) => {    
    setInputValue( e.target.value );
  }

  const handleSubmit = ( e ) => {
    e.preventDefault();
    if(inputValue.trim().length>2){
      setCategories( cats => [...cats,inputValue] );
      setInputValue('');
    }    
  }
 
  return (
    <>      
      <form onSubmit={ handleSubmit }>
        <input
          type="text"
          value={ inputValue }
          onChange={ handleInputChange }
        />
      </form>
      <button onClick={ handleAdd }>agrega</button>
    </>
  )
}
