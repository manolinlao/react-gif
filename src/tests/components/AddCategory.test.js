// este componente recibe una funci贸n que se ha de llamar cuando el input tiene valor
// llamando al submit( handleSubmit )

import React from 'react';
import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";


describe( 'Pruebas en <AddCategory/>', () => {

  // Esta funci贸n la proporciona Jest
  const setCategories = jest.fn();

  let wrapper = shallow( <AddCategory setCategories={ setCategories }/> );

  beforeEach( () => {
    jest.clearAllMocks();
    wrapper = shallow( <AddCategory setCategories={ setCategories }/> );
  });

  // Prueba 1 - El componente se ha de mostrar correctamente
  test( 'debe de mostrarse correctamente', () => {    
    expect( wrapper ).toMatchSnapshot();
  });

  // Prueba 2 - vamos a evaluar cuando la caja de texto cambie
  test( 'debe de cambiar la caja de texto', () => {    
    const input = wrapper.find('input');
    const value = 'hola mundo';
    input.simulate( 'change',{ target:{ value } } );
  });

  // Prueba 3 - un submit del form no ha de llamar a setCategories
  test( 'no debe postear informaci贸n con submit', () => {
    wrapper.find( 'form').simulate( 'submit' , { preventDefault:() => {} } );
    expect( setCategories ).not.toHaveBeenCalled();
  });

  // Prueba 4 - un submit de un value de length<3 no ha de llamar a setCategories
  test( 'no debe postear informaci贸n con submit de value.length<3', () => {
    const input = wrapper.find('input');
    input.simulate( 'change',{ target:{ value:'ab' } } );
    wrapper.find( 'form').simulate( 'submit' , { preventDefault:() => {} } );
    expect( setCategories ).not.toHaveBeenCalled();
  });

   // Prueba 4 - un submit de un value de length>2 ha de llamar a setCategories y limpiar la caja de texto
   test( 'debe llamar a setCagegories y limpair la caja de texto', () => {
    let input = wrapper.find( 'input' );
    input.simulate( 'change',{ target:{ value:'abc' } } );
    wrapper.find( 'form').simulate( 'submit' , { preventDefault:() => {} } );
    expect( setCategories ).toHaveBeenCalled();
   
    input = wrapper.find('input');
    expect( input.prop( 'value' ) ).toBe( '' );
  });

});