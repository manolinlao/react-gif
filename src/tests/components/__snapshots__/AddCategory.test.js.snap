// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory/> debe de mostrarse correctamente 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
  <button
    onClick={[Function]}
  >
    agrega
  </button>
</Fragment>
`;
